src/Umbraco.Infrastructure/DependencyInjection/UmbracoBuilder.CoreServices.cs

```diff
@@ -248,6 +248,10 @@ public static IUmbracoBuilder AddCoreInitialServices(this IUmbracoBuilder builde
 
         builder.Services.AddSingleton<IRichTextRequiredValidator, RichTextRequiredValidator>();
 
+        // same as newing one up and required for the RichTextRegexValidator
+        builder.Services.AddTransient<RegexValidator>();
```

Comment:
Maybe remove this and just new it up in `RichTextRegexValidator`?  That'll happen anyway as it's registered transient, and aligns with how it's used in `DataValueEditor.